### ISSUE TITLE
tasks.list not working

### ISSUE THREAD

**Issue by odelucca** [simplyspoke/node-harvest#83]
Hi.

I am trying to call the tasks.list but it isn't working. I was debbuging the code and it isn't even calling the list function from the tasks helper.

The code that I have written is:
```
exports.extractTasks = () => {
    let promise = new Promise((resolve, reject) => {
        harvest.tasks.list({}, (error, res, body) => {
            resolve(body);
        });
    });

    return promise;
}

```
Can anyone help me?

**Comment by simplyspoke** [comment]
I can't promise to resolve your issue, but I won't reject your ask for help. Right... think that want cost me my joking privileges for a week.

Just ran the tests and was able to log a list of tasks successfully to screen. Here are a couple of thoughts:

- It is a good idea to a catch setup for possible error handling. Might save you some debug in the long run.
- The promise variable ins't truly needed in this case as you are immediately returning it. Might be better to think of this as a generator function.
- How are you calling your function? Could it be something there?

Here is a modified version of your code that would also handle errors:
```
exports.extractTasks = (options = {}) => {
  return new Promise((resolve, reject) => {
    harvest.tasks.list(options, (error, res, body) => {
      if (error) {
        reject(error);
      }
      resolve(body);
    });
  });
}
```
The above also would allow you to pass in parameters or options should you need, but has a default of an empty object. Then for your call:
```
extractTasks()
  .then((results) => {
    // do some stuff
  })
  .catch((error) => {
    // handle your error
  })
```

Does any of that help?

**Comment by odelucca** [comment]
@simplyspoke Unfortunately no.

I added the error handling at the function but it still isn't working. I tried to add some simple console logs in the harvest original module, at the list function, just to output any info if the tasks.list is called, but It is outputting the info at every other resource but not in the tasks resource.

More details: 

I changed the harvest > lib > mixins > api-base.js list function to:

```
  list(options, cb) {
    let uri = this.baseUri;
    
    console.log(uri) //simple debug
    this.harvest.request('GET', uri, options, cb);
  },
```

When I run the code, the order of the execution are: extract clients - extract tasks extract users - extract projects

I got these outputs:
```
/clients/
and nothing more
```

If I remove the extract tasks from the queu I got:
```
/clients/
/people/
/projects/
```

It seems like it is getting stucked while trying to list all the tasks
